sql_generators/glean_usage/templates/dau_reporting_clients_daily_v1.query.sql

```diff
@@ -0,0 +1,143 @@
+{{ header }}
+
+WITH base AS (
+  SELECT
+    submission_timestamp,
+    DATE(submission_timestamp) AS submission_date,
+    client_info.client_id,
+    sample_id,
+    normalized_channel,
+    client_info.app_display_version,
+    client_info.app_build,
+    normalized_os,
+    normalized_os_version,
+    client_info.locale,
+    -- metadata.geo.country,
+    {% if has_distribution_id %}
+    metrics.string.metrics_distribution_id AS distribution_id,
+    {% else %}
+    CAST(NULL AS STRING) AS distribution_id,
+    {% endif %}
+    {% if has_profile_group_id %}
+    metrics.uuid.legacy_telemetry_profile_group_id AS profile_group_id,
+    {% else %}
+    CAST(NULL AS STRING) AS profile_group_id,
+    {% endif %}
+    {% if app_name == "firefox_desktop" %}
+    metrics.counter.browser_engagement_uri_count,
+    metrics.counter.browser_engagement_active_ticks,
+    metrics.uuid.legacy_telemetry_client_id,
+    {% endif %}
+    SAFE.PARSE_DATE('%F', SUBSTR(client_info.first_run_date, 1, 10)) AS first_run_date,
+    mozfun.glean.parse_datetime(ping_info.end_time) AS parsed_end_time,
+    -- udf.glean_timespan_seconds(metrics.timespan.glean_baseline_duration) AS duration, -- TODO: duration appears to be missing in the ping?
+  FROM
+    `{{ project_id }}.{{ dau_reporting_stable_table }}`
+),
+--
+with_dates AS (
+  SELECT
+    *,
+    -- For explanation of session start time calculation, see Glean docs:
+    -- https://mozilla.github.io/glean/book/user/pings/baseline.html#contents
+    DATE(SAFE.TIMESTAMP_SUB(parsed_end_time, INTERVAL duration SECOND)) AS session_start_date,
```

Comment:
I am wondering if the fields from the `with_dates` and `with_date_offsets` ctes could be included in the `base` cte itself?

Also, `duration` used in `session_start_date` is missing in the `base` cte

Reply:
Yes, that should not be there yet. My understanding is that `duration` is missing at the moment from the ping and is something that we will need to add to the logic once it is there.